lnwallet/interface_test.go

```diff
@@ -786,11 +787,62 @@ func testListTransactionDetails(miner *rpctest.Harness,
 			t.Fatalf("tx value incorrect, got %v expected %v",
 				txDetail.Value, outputAmt)
 		}
+
 		if !bytes.Equal(txDetail.BlockHash[:], blocks[0][:]) {
 			t.Fatalf("block hash mismatch, got %v expected %v",
 				txDetail.BlockHash, blocks[0])
 		}
 
+		// This fetches the transactions in a block so that we can compare the
+		// txouts stored in the mined transaction against the ones in the transaction
+		// details
+		if _, ok := blockTxOuts[*txDetail.BlockHash]; !ok {
+			fetchedBlock, err := alice.Cfg.ChainIO.GetBlock(txDetail.BlockHash)
+			if err != nil {
+				t.Fatalf("err fetching block: %s", err)
+			}
+
+			transactions := make(map[chainhash.Hash][]*wire.TxOut, len(fetchedBlock.Transactions))
+			for _, tx := range fetchedBlock.Transactions {
+				transactions[tx.TxHash()] = tx.TxOut
+			}
+
+			blockTxOuts[fetchedBlock.BlockHash()] = transactions
+		}
+
+		if txOuts, ok := blockTxOuts[*txDetail.BlockHash][txDetail.Hash]; !ok {
+			t.Fatalf("tx (%v) not found in block (%v)", txDetail.Hash, txDetail.BlockHash)
+		} else {
+			var (
+				destinationAddresses []btcutil.Address
+				totalAddresses       int
+			)
+
+			for _, txOut := range txOuts {
+				_, addrs, addressCount, err := txscript.ExtractPkScriptAddrs(txOut.PkScript, &alice.Cfg.NetParams)
+				if err != nil {
+					t.Fatalf("err extract script addresses: %s", err)
+				}
+				destinationAddresses = append(destinationAddresses, addrs...)
+				totalAddresses += addressCount
+			}
+
+			if len(txDetail.DestinationAddresses) != len(destinationAddresses) {
+				t.Fatalf("expected %v destination address, got %v ", len(txDetail.DestinationAddresses), len(destinationAddresses))
```

Comment:
We typically try on a best-effort basis to wrap to 80 characters. This diff has a few offenders.